lib/src/rules/directives_ordering.dart

```diff
@@ -83,10 +124,24 @@ class _Visitor extends SimpleAstVisitor {
 
   @override
   void visitCompilationUnit(CompilationUnit node) {
+    _checkDartImportGoFirst(node);
+    _checkPackageImportBeforeRelative(node);
+  }
```

Comment:
You might consider taking a different approach here and write the whole rule as a state machine that could make a single pass over the directives. This approach might work fine, but it also might be difficult to prevent generating extraneous lints after you've added logic about how to order `package:` URIs and the sort order logic.

Alternatively, you could take a two pass approach where the first pass collects imports into groups, flagging those that appear in the wrong group, and then a second pass that flags issues related to ordering within each group.

As a bigger question, would the UX be better if we generated a single lint when the imports are out of order that pointed users to the guidelines, rather than producing potentially large numbers of lints? Maybe most users won't see very many lints and it is better to have more specific error messages; I honestly don't know.

Have you run this lint over any existing code bases to see what kind of output it produces? If not, we should do that before enabling the rule in general.